Feature/dual module support

zkabyken opened the pull request:
## Description

Adds CommonJS build support alongside existing ES module support, enabling the library to work in both CommonJS and ES module environments while maintaining full backward compatibility. This enhancement allows the library to be used in a wider range of Node.js environments and build systems

The changes include:
- Added `"require"` export and updated `"main"` entry point in package.json
- Added `build:cjs` and `build:all` scripts for CommonJS compilation
- Created `tsconfig.cjs.json` for CommonJS TypeScript configuration
- Added `dist-cjs/package.json` with `"type": "commonjs"` for proper module resolution
- Moved `index.ts` to `src/index.ts` for better code organization

Big fan of the project and would love to see it succeed! The dual module support will make the library more accessible to different development environments and build systems.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] The title of my pull request follows the [conventional commits](https://www.conventionalcommits.org/) standard
- [x] Changes have been documented in the README/documentation (if applicable)

Pederzh commented:
> Adds CommonJS build support alongside existing ES module support, enabling the library to work in both CommonJS and ES module environments while maintaining full backward compatibility. This enhancement allows the library to be used in a wider range of Node.js environments and build systems
> 
> The changes include:
> 
> * Added `"require"` export and updated `"main"` entry point in package.json
> * Added `build:cjs` and `build:all` scripts for CommonJS compilation
> * Created `tsconfig.cjs.json` for CommonJS TypeScript configuration
> * Added `dist-cjs/package.json` with `"type": "commonjs"` for proper module resolution
> * Moved `index.ts` to `src/index.ts` for better code organization
> 
> Big fan of the project and would love to see it succeed! The dual module support will make the library more accessible to different development environments and build systems.

Hi @zkabyken we're glad that you are a fan of the project, thanks for your support :)

Thanks for taking the time to create this PR! 
I've made some changes to the library. 
Can you please resolve the merge conflicts?

We'd love to have you join our Discord community to chat about how you're planning to use mcp-use!
https://discord.gg/XkNkSkMz3V

Pederzh commented on package.json:
```diff
@@ -46,6 +48,8 @@
   },
   "scripts": {
     "build": "rm -rf dist && tsc",
+    "build:cjs": "rm -rf dist-cjs && tsc --project tsconfig.cjs.json && echo '{\"type\":\"commonjs\"}' > dist-cjs/package.json",
```
@zkabyken, can you also have a look at this please?

zkabyken commented:
Hi @Pederzh, I’ve addressed all the comments, this PR should be fully ready now. Let me know if there’s anything else you’d like me to tweak. Super excited to see it get merged!

zkabyken commented:
Hi @pietrozullo! Wondering if you could take a look at this PR whenever possible, would love to continue contributing after this is merged!
